File: Data-Structure-I/101-Symmetric-Tree.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return Boolean
     */
    function isSymmetric($root) {
        if(!$root) return true;
        return $this->check($root->left, $root->right);
    }
    
    function check($left, $right){
        if(!$left || !$right) return $left == $right;
        if($left->val != $right->val) return false;
        return $this->check($left->left, $right->right) && $this->check($left->right, $right->left);
    }
}

File: Data-Structure-I/101-Symmetric-Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return true
        return self.check(root.left, root.right)
    
    def check(self, left: Optional[TreeNode], right: Optional[TreeNode]) -> bool:
        if not left or not right:
            return left==right
        if left.val != right.val:
            return False
        return self.check(left.left, right.right) and self.check(left.right, right.left)
